src/core/dbus-manager.c

```diff
@@ -2783,6 +2783,50 @@ static int method_set_show_status(sd_bus_message *message, void *userdata, sd_bu
         return sd_bus_reply_method_return(message, NULL);
 }
 
+/* Create a new user namespace that maps all UIDs/GIDs in the system and return the
+ * file descriptor to the caller. */
+static int method_get_uid_gid_mapping(sd_bus_message *message, void *userdata, sd_bus_error *error) {
+        _cleanup_(sd_bus_creds_unrefp) sd_bus_creds *creds = NULL;
+        _cleanup_free_ char *line = NULL;
+        _cleanup_close_ int fd = -1;
+        Manager *m = ASSERT_PTR(userdata);
+        uid_t user;
+        int r;
+
+        assert(message);
+
+        if (!MANAGER_IS_SYSTEM(m))
+                return sd_bus_error_setf(error, SD_BUS_ERROR_NOT_SUPPORTED,
+                                         "This method is only supported on the system manager.");
+
+        r = mac_selinux_access_check(message, "acquire-namespace", error);
+        if (r < 0)
+                return r;
+
+        r = bus_verify_acquire_namespace_async(m, message, error);
+        if (r < 0)
+                return r;
+        if (r == 0)
+                return 1; /* No authorization for now, but the async polkit stuff will call us again when it has it */
+
+        r = sd_bus_query_sender_creds(message, SD_BUS_CREDS_UID|SD_BUS_CREDS_AUGMENT, &creds);
```

Comment:
drop SD_BUS_CREDS_AUGMENT. that's the flag you specify if you want stuff to be added from /proc/. But you don#t wnat that here, and it's not suitable for security decisions like yours here.

Reply:
I need the pid, and for that SD_BUS_CREDS_AUGMENT is needed, I do not get it without this

Reply:
if you want the PID use SD_BUS_CREDS_PID?

or do you mean to say UID? if so, does it give you the EUID without augment?